samcli/lib/providers/sam_function_provider.py

```diff
@@ -387,8 +362,25 @@ def _parse_layer_info(
             # In the list of layers that is defined within a template, you can reference a LayerVersion resource.
             # When running locally, we need to follow that Ref so we can extract the local path to the layer code.
             if isinstance(layer, dict) and layer.get("Ref"):
-                layer_logical_id = cast(str, layer.get("Ref"))
-                layer_resource = stack.resources.get(layer_logical_id)
+                # Here layer refers to another resources.
+                # in a multi stack system, the layer might locate in the current stack or passed down from parents.
+                # for example, {"Ref": "layerLogicalID"} refers to the "layerLogicalID" in the current stack.
+                # {"Ref": {"CrossStackRef": ["..", "layerLogicalID"]}
```

Comment:
Wouldn't adding a new SAM CLI specific intrinsic function break existing translation? The template will not be compatible with uploading to CFN directly.

Reply:
The built template won't contain this special function. 

The output template because it is only affected by `builder.update_template()`(only update codeuri and other path properties)  and `template.move_template()` (only move the file).